addons/website/models/website_visitor.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, api
+from odoo.http import request
+from datetime import datetime, timedelta
+from hashlib import md5
+
+
+class WebsiteVisitor(models.Model):
+    _name = 'website.visitor'
+    _description = 'Website Visitor'
+    _order = 'last_connection_date DESC'
+
+    name = fields.Char('Name', default='Website Visitor')
+    website_id = fields.Many2one('website', "Website", readonly=True)
+    partner_ids = fields.Many2many('res.partner')
```

Comment:
I would rename to user_partner_ids and add an help to explain we have a m2m relationships between user's partners and visitors, because of shared visits.